packages/tools/kadena-cli/README.md

```diff
@@ -716,27 +736,121 @@ Tool for creating and managing transactions
+| `--holes`               | Displays a list of required template variables.           | No       |
+| Custom options          | Generated based on the chosen template's required fields. | Varies   |
+
+### Example Command
+
+```
+kadena tx add --template="transfer.yaml" --template-data="data.yaml" (--account-from="k:account" --account-to="k:toaccount" --amount="1" --chain-id="1" --pk-from="publicKey") --network-id="testnet04" --out-file="transaction.json"
+```
+
+In this example, `transfer.yaml` is the template used to construct the
+transaction. `data.yaml` contains the user-supplied values for the template
+variables. The `--network-id` specifies which network the transaction is
+intended for, and `--out-file` determines where the generated transaction file
+will be saved.
+
+### Template Variables
+
+Variables are a critical part of transaction templates, defining the data
+required to construct a transaction. Users can be prompted for variables missing
+from the `--template-data` file or not provided as command-line arguments. The
+`--holes` option is particularly useful for identifying all the variables a
+template requires.
+
+Variables support specific prefixes (`account:`, `key:`, `network:`, `decimal:`)
+to facilitate the correct selection or validation of input values in interactive
+mode.
+
+`account`: This prefix is used for variables that should be filled with an
+account name, guiding users to input valid Kadena account identifiers. `key`:
+Variables with this prefix expect a public key, ensuring that users provide
+cryptographic keys in the correct format. `network`: This prefix is used for
+specifying the network ID, helping users to select the appropriate network for
+their transaction. `decimal`: For variables that involve numerical values with
+decimal points, this prefix ensures the format and precision of the input.
+
+Below is a YAML template that outlines the structure for a coin transfer
+operation on Kadena. Notice the use of placeholders with prefixes to define
+expected data types for each field:
+
+```yaml
+code: |-
+  (coin.transfer "{{{account:from}}}" "{{{account:to}}}" {{decimal:amount}}))
+  (coin.transfer "{{{account:to}}}" "{{{account:from}}}" 0.000000000001)
+data:
+meta:
+  chainId: '{{chain-id}}'
+  sender: { { { account:from } } }
+  gasLimit: 2000
+  gasPrice: 0.00000001
+  ttl: 7200
+signers:
+  - public: { { key:from } }
+    caps:
+      - name: 'coin.TRANSFER'
+        args:
+          [
+            { { { account:from } } },
+            { { { account:to } } },
+            { { decimal:amount } },
+          ]
+      - name: 'coin.GAS'
+        args: []
+  - public: { { key:to } }
+    caps:
+      - name: 'coin.TRANSFER'
+        args: [{ { { account:to } } }, { { { account:from } } }, 0.000000000001]
+networkId: '{{network:networkId}}'
+type: exec
+```
+
+This template exemplifies how prefixes guide users in providing the correct
```

Comment:
(just question) do we also want to mention about where default templates are stored ie. `.kadena/transaction-templates`

Reply:
yeah i like that, no magic idd lets be specific, ill add it